### PR TITLE
fix(ESSNTL-5418): Trim group name before validate

### PR DESCRIPTION
**Pull request by gkarat**
Fixes https://issues.redhat.com/browse/ESSNTL-5418.

## How to test

1. Go to /groups and click create a new group
2. Try to add a name of existing group and also add some whitespaces before/after the name
3. The modal should not allow you to create a group

### CODE REVIEW

File: src/components/InventoryGroups/Modals/CreateGroupModal.js [hunk 1]
```diff
@@ -7,6 +7,15 @@ import { createGroup, validateGroupName } from '../utils/api';
 import { useDispatch } from 'react-redux';
 import awesomeDebouncePromise from 'awesome-debounce-promise';
 
+export const validate = async (value) => {
+  const results = await validateGroupName(value.trim());
+  if (results === true) {
+    throw 'Group name already exists';
+  }
+
+  return undefined;
```
Comment (mkholjuraev) [inline src/components/InventoryGroups/Modals/CreateGroupModal.js:16]: Question: why would we need to return undefined explicitly?

**Review by mkholjuraev — APPROVED** [review]
LGTM! I have tried creating groups that had white space before and after, also with existing name. Modal did not allow me. Nice work!

File: src/components/InventoryGroups/Modals/CreateGroupModal.js (hunk 1 above)
Comment (gkarat) [reply src/components/InventoryGroups/Modals/CreateGroupModal.js:16]: This is the requirement of data-driven-forms. In order to mark that the validator succeeds, it shouldn't return nothing or just undefined 😄

**Pull request merged**